src/routes/+layout.svelte

```diff
@@ -0,0 +1,190 @@
+
+	let topAppBar: TopAppBar;
+	let surface: MenuSurface;
+	let drawerOpened = false;
+	$: menuIcon = drawerOpened ? 'menu_open' : 'menu';
+
+	// Navigation to profile page
+	function navigateToProfile() {
+		// Prevent navigation if already on profile page
+		if (window.location.pathname.includes('profile')) return;
+		// TO-DO: Replace with correct username
+		goto('profile/test', { replaceState: false });
+	}
+
+	// Theme related variables and functions
+	let lightTheme =
+		typeof window === 'undefined' || window.matchMedia('(prefers-color-scheme: light)').matches;
+	$: themeIcon = lightTheme ? 'light_mode' : 'dark_mode';
+	$: themeLabel = lightTheme ? 'Switch to dark theme' : 'Switch to light theme';
+
+	function switchTheme() {
+		lightTheme = !lightTheme;
+		let themeLink = document.head.querySelector<HTMLLinkElement>('#theme');
+
+		if (!themeLink) {
+			themeLink = document.createElement('link');
+			themeLink.rel = 'stylesheet';
+			themeLink.id = 'theme';
+		}
+		themeLink.href = `/smui${lightTheme ? '' : '-dark'}.css`;
+		document.head
+			.querySelector<HTMLLinkElement>('link[href$="/smui-dark.css"]')
+			?.insertAdjacentElement('afterend', themeLink);
+	}
+</script>
+
+<TopAppBar bind:this={topAppBar} variant="standard" class="navbar">
+	<Row>
+		<Section>
+			<IconButton
+				class="material-icons"
+				aria-label="menu_icon"
+				on:click={() => (drawerOpened = !drawerOpened)}>{menuIcon}</IconButton
+			>
+			<Title>Expense Management</Title>
+		</Section>
+		<Section align="end" toolbar>
+			<IconButton class="material-icons" aria-label="theme_icon" on:click={switchTheme}
+				>{themeIcon}
+			</IconButton>&nbsp;{themeLabel}
+			<div class="accountOverlayWrapper">
+				<IconButton
+					class="material-icons"
+					aria-label="account_icon"
+					on:click={() => surface.setOpen(true)}
+					>account_circle
+				</IconButton>
+				<MenuSurface bind:this={surface} anchorCorner="BOTTOM_LEFT">
+					<Card>
+						<PrimaryAction on:click={navigateToProfile}>
+							<Media class="card-media-16x9" aspectRatio="16x9" />
+							// TO-DO: Replace with correct data
+							<MediaContent class="mdc-typography--body2">
+								<h2 class="mdc-typography--headline6" style="margin: 0;">Insert Username</h2>
+							</MediaContent>
+							<CardContent>
+								<h3 class="mdc-typography--subtitle2" style="margin: 0 0 10px; color: #888;">
+									And some other information
+								</h3>
+							</CardContent>
+						</PrimaryAction>
+						<Actions>
+							<ActionButtons>
+								<Button on:click={navigateToProfile}>
+									<Label>Profile</Label>
+								</Button>
+							</ActionButtons>
+							<ActionIcons>
+								<IconButton class="material-icons" on:click={() => goto('/settings')}
+									>settings</IconButton
+								>
+								<IconButton class="material-icons" on:click={() => goto('/logout')}
+									>logout</IconButton
+								>
+							</ActionIcons>
+						</Actions>
+					</Card>
```

Comment:
Wäre das hier nicht eine eigene Komponente?

Reply:
Machts übersichtlicher, kann ich auslagern, ja

Reply:
> Wir haben uns gerade gefragt wo wir script, style und html hinpacken sollen, damit die Files einheitlich sind. Die SMUI Components machen ja das Script immer unters HTML aber Svelte macht das immer oben drüber. Auf was wollen wir uns da einigen?

Ich würde das so machen wie bei mir jetzt:

<script lang="ts"> ... </script>

content

<style> ... </style>

Der TypeScript-Teil ist imo der wichtigste Teil und sollte oben stehen

Reply:
Habs in ne Komponente ausgelagert und gleich mal ne Ordnerstruktur vorgegeben, würde Komponenten unter `src/components` ablegen.

Ich hab zusätzlich einen alias angelegt in unserer Config, dadurch kann man Komponenten jetzt mit `import {.componentXY } from '$components'` importieren. Vorausgesetzt man exportiert die Komponente davor in `src/components/index.ts` mit `export {default as componentXY} from './componentXY'`